relayer/src/chain/runtime.rs

```diff
@@ -1,71 +1,542 @@
-use crate::chain::handle::{cosmos::CosmosSDKHandle, ChainHandleError, HandleInput};
-use crate::config::ChainConfig;
-use crate::msgs::{Datagram, EncodedTransaction, IBCEvent, Packet};
-
-use ibc::Height;
+use std::{sync::Arc, thread};
 
 use crossbeam_channel as channel;
-use std::time::Duration;
 
-pub struct ChainRuntime {
-    chain_config: ChainConfig,
+use tokio::runtime::Runtime as TokioRuntime;
+
+use ibc::{
+    ics02_client::{
+        client_def::{AnyClientState, AnyConsensusState, AnyHeader},
+        header::Header,
+        state::{ClientState, ConsensusState},
+    },
+    ics03_connection::connection::ConnectionEnd,
+    ics23_commitment::{commitment::CommitmentPrefix, merkle::MerkleProof},
+    ics24_host::identifier::{ClientId, ConnectionId},
+    ics24_host::Path,
+    proofs::Proofs,
+    Height,
+};
+
+// FIXME: the handle should not depend on tendermint-specific types
+use tendermint::account::Id as AccountId;
+
+// use crate::foreign_client::ForeignClient;
+
+use crate::{
+    config::ChainConfig,
+    error::{Error, Kind},
+    event::{
+        bus::EventBus,
+        monitor::{EventBatch, EventMonitor},
+    },
+    keyring::store::KeyEntry,
+    light_client::{tendermint::LightClient as TMLightClient, LightClient},
+    tx::connection::ConnectionMsgType,
+};
+
+use super::{
+    handle::{ChainHandle, HandleInput, ProdChainHandle, ReplyTo, Subscription},
+    Chain, CosmosSDKChain, QueryResponse,
+};
+
+pub struct Threads {
+    pub light_client: thread::JoinHandle<()>,
+    pub chain_runtime: thread::JoinHandle<()>,
+    pub event_monitor: thread::JoinHandle<()>,
+}
+
+pub struct ChainRuntime<C: Chain> {
+    chain: C,
     sender: channel::Sender<HandleInput>,
     receiver: channel::Receiver<HandleInput>,
+    event_bus: EventBus<Arc<EventBatch>>,
+    event_receiver: channel::Receiver<EventBatch>,
+    light_client: Box<dyn LightClient<C>>,
+
+    #[allow(dead_code)]
+    rt: Arc<TokioRuntime>,
```

Comment:
It's not necessary at all at the moment (hence the need for suppressing the warning). I can't really justify why it's there except that it felt right/future-proof but since that can indeed be confusing, we can totally remove it.

Reply:
I'm trying to understand all the threads that are running, so I tracked down this `rt` runtime and this was a dead-end, hence my question.

But let's keep it around then. You have a good point it may likely be used in future -- but maybe add a comment to inform others that it's not used at the moment, yet we keep it around for "future-proof reasons" ?